llvm/lib/Analysis/IVDescriptors.cpp

```diff
@@ -721,9 +723,15 @@ RecurrenceDescriptor::isAnyOfPattern(Loop *Loop, PHINode *OrigPhi,
 //     if (src[i] > 3)
 //       r = i;
 //   }
+// or like this:
+//   int r = 0;
+//   for (int i = 0; i < n; i++) {
+//     if (src[i] > 3)
+//       r = src[i];
```

Comment:
```suggestion
//       r = src[i];
```

IIUC there are no restrictions here, perhaps better use something more generic for the term here. Also the condition does not have to be related to the value selected value